crates/router/src/core/connector_validation.rs

```diff
@@ -0,0 +1,581 @@
+            }
+            api_enums::Connector::Wellsfargo => {
+                wellsfargo::transformers::WellsfargoAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Wise => {
+                wise::transformers::WiseAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Worldline => {
+                worldline::transformers::WorldlineAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Worldpay => {
+                worldpay::transformers::WorldpayAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Worldpayvantiv => {
+                worldpayvantiv::transformers::WorldpayvantivAuthType::try_from(self.auth_type)?;
+                worldpayvantiv::transformers::WorldpayvantivMetadataObject::try_from(
+                    self.connector_meta_data,
+                )?;
+                Ok(())
+            }
+            api_enums::Connector::Worldpayxml => {
+                worldpayxml::transformers::WorldpayxmlAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Xendit => {
+                xendit::transformers::XenditAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Zen => {
+                zen::transformers::ZenAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Zsl => {
+                zsl::transformers::ZslAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Signifyd => {
+                signifyd::transformers::SignifydAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Riskified => {
+                riskified::transformers::RiskifiedAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Plaid => {
+                PlaidAuthType::foreign_try_from(self.auth_type)?;
+                Ok(())
+            }
+            api_enums::Connector::Threedsecureio => {
+                threedsecureio::transformers::ThreedsecureioAuthType::try_from(self.auth_type)?;
+                Ok(())
+            }
+        }
+    }
+}
+
+struct ConnectorAuthTypeValidation<'a> {
+    auth_type: &'a types::ConnectorAuthType,
+}
+
+impl ConnectorAuthTypeValidation<'_> {
+    fn validate_connector_auth_type(
+        &self,
+    ) -> Result<(), error_stack::Report<errors::ApiErrorResponse>> {
+        let validate_non_empty_field = |field_value: &str, field_name: &str| {
+            if field_value.trim().is_empty() {
+                Err(errors::ApiErrorResponse::InvalidDataFormat {
+                    field_name: format!("connector_account_details.{}", field_name),
```

Comment:
Clippy 1.88 would mostly throw a warning here as well.

```suggestion
                    field_name: format!("connector_account_details.{field_name}"),
```

Reply:
fixed